addons/link_tracker/models/link_tracker.py

```diff
@@ -197,6 +196,27 @@ def search_or_create(self, vals):
 
         return self.create(vals)
 
+    @api.model
+    def search_or_create_multi(self, vals_list):
+        for vals in vals_list:
+            if 'url' not in vals:
+                raise ValueError(_('Creating a Link Tracker without URL is not possible'))
+            vals['url'] = tools.validate_url(vals['url'])
+
+        search_domains = [[
+            (fname, '=', value)
+            for fname, value in vals.items()
+            if fname in {'url', 'campaign_id', 'medium_id', 'source_id', 'label'}
+        ] for vals in vals_list]
+
+        search_domain = expression.OR(search_domains)
+        result = self.search(search_domain)
+
+        if not result:
+            return self.create(vals_list)
+
+        raise NotImplemented(f"Found {len(result)} results: {result}")
```

Comment:
Can't you rebuild based on vals_list (from search or from create) ? A bit more complex but would allow a real multi search or create :)